app/src/main/java/com/opsat7373/moviestars/data/datasource/WebDataSource.kt

```diff
@@ -0,0 +1,39 @@
+package com.opsat7373.moviestars.data.datasource
+
+import androidx.lifecycle.MutableLiveData
+import com.opsat7373.moviestars.data.model.Movie
+import com.opsat7373.moviestars.data.model.PopularMoviesList
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import java.util.*
+
+
+class WebDataSource : MovieDataSourceInterface {
+    override fun getAll(): MutableLiveData<List<Movie?>> {
+        val retrofit = Retrofit.Builder()
+            .baseUrl("https://api.themoviedb.org/")
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
+        val service = retrofit.create(TMDBService::class.java)
```

Comment:
Pass the service as an external dependency using a constructor.
No need to build a retrofit client for each call.